### PR TITLE
toggle security usage

### PR DESCRIPTION
**Pull request by MouazAlzahabi**
Handles #2825

### CODE REVIEW

File: Multisig/Features/Security/SecurityCenter.swift [hunk 1]
```diff
@@ -163,6 +163,49 @@ class SecurityCenter {
         try changeStoreSettings(currentPlaintextPassword: oldPasscode, newPlaintextPassword: newPasscode, store: dataStore)
     }
 
+    func toggleUsage(passcodeOption: PasscodeOptions, completion: @escaping (Error?) -> Void) {
+        let enabledOptions = AppSettings.passcodeOptions
+
+        let isBothTogglesWillBeOff = AppSettings.passcodeOptions.contains(passcodeOption) &&
```
Comment (elgatovital) [inline Multisig/Features/Security/SecurityCenter.swift]: rename -> willBothTogglesBeOff?

Comment (elgatovital) [reply Multisig/Features/Security/SecurityCenter.swift]: or bothTogglesWillBeOff. Then if statement will be read more naturally

File: Multisig/Features/Security/SecurityCenter.swift [hunk 2]
```diff
@@ -163,6 +163,49 @@ class SecurityCenter {
...
+
+        if bothTogglesWillBeOff {
+            disableSecurityLock(completion: completion)
+            return
+        }
+
+        let protectionClass: ProtectionClass = passcodeOption == .useForLogin ? .data : .sensitive
+        let protectedKeyStore: ProtectedKeyStore = passcodeOption == .useForLogin ? dataStore : sensitiveStore
+
+        let isToggleWillBeOff: Bool = enabledOptions.contains(passcodeOption)
```
Comment (elgatovital) [inline Multisig/Features/Security/SecurityCenter.swift]: I would rename this as well: -> toggleWillBeOff

**Review by elgatovital — APPROVED** [review]

File: Multisig/Features/Security/SecurityCenter.swift [hunk 3]
```diff
@@ -163,6 +163,49 @@ class SecurityCenter {
...
+            } onFailure: { error in
+                AppSettings.passcodeOptions.insert(passcodeOption)
+                completion(error)
+            }
+        } else {
+            let otherProtectionClass: ProtectionClass = protectionClass == .sensitive ? .data : .sensitive
+
+            requestPasswordV2(for: [otherProtectionClass]) { [unowned self] plaintextPasscode in
+                AppSettings.passcodeOptions.insert(passcodeOption)
+                try changeStoreSettings(currentPlaintextPassword: nil,
```
Comment (DmitryBespalov) [inline Multisig/Features/Security/SecurityCenter.swift:198]: did you decide to skip the authentication in the end?

Comment (MouazAlzahabi) [reply Multisig/Features/Security/SecurityCenter.swift:198]: No, I will upload latest changes

**Pull request merged**